data-prepper-plugins/rds-source/src/test/java/org/opensearch/dataprepper/plugins/source/rds/datatype/impl/TemporalTypeHandlerTest.java

```diff
@@ -1,32 +1,182 @@
-import org.opensearch.dataprepper.plugins.source.rds.model.TableMetadata;
 
-import java.util.Collections;
-import java.util.List;
-import java.util.UUID;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.ZoneOffset;
+import java.util.Date;
 
-import static org.hamcrest.CoreMatchers.instanceOf;
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
-public class TemporalTypeHandlerTest {
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+class TemporalTypeHandlerTest {
+    private TemporalTypeHandler temporalTypeHandler;
+
+    @BeforeEach
+    void setUp() {
+        temporalTypeHandler = new TemporalTypeHandler();
+    }
+
+    private static long getEpochMillisFromDateStr(final String dateStr) throws ParseException {
+        final SimpleDateFormat mysqlDateFormat = new SimpleDateFormat("yyyy-MM-dd");
+
+        final Date date = mysqlDateFormat.parse(dateStr);
+        return date.getTime();
+    }
+
+    private static long getEpochMillis(int year, int month, int day, int hour, int minute, int second, int nanoSeconds) {
+        return LocalDateTime.of(year, month, day, hour, minute, second, nanoSeconds)
+                .toInstant(ZoneOffset.UTC)
+                .toEpochMilli();
+    }
+
+    @Test
+    void handle_whenValueIsNull_returnsNull() {
+        assertNull(temporalTypeHandler.handle(MySQLDataType.DATE, "test_column", null, null));
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideDateTestCases")
+    void handle_withDateType_returnsCorrectEpochMillis(String input, long expected) {
+        Long result = temporalTypeHandler.handle(MySQLDataType.DATE, "date_column", input, null);
+        assertEquals(expected, result);
+    }
+
+    private static Stream<Arguments> provideDateTestCases() throws ParseException {
+        return Stream.of(
+                Arguments.of("2023-12-25", getEpochMillisFromDateStr("2023-12-25")),
+                Arguments.of("1970-01-01", getEpochMillisFromDateStr("1970-1-1")),
+                Arguments.of("2024-02-29", getEpochMillisFromDateStr("2024-2-29")) // Leap year
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideTimeTestCases")
+    void handle_withTimeType_returnsCorrectEpochMillis(String input, long expected) {
+        Long result = temporalTypeHandler.handle(MySQLDataType.TIME, "time_column", input, null);
+        assertEquals(result, expected);
+    }
+
+    private static Stream<Arguments> provideTimeTestCases() {
+        return Stream.of(
```

Comment:
Similarly, we may want to add a test case with a string of long value as input.

Reply:
Added